pyat/at/load/utils.py

```diff
@@ -139,28 +148,36 @@ def sanitise_class(index, class_name, kwargs):
             AttributeError: if the PassMethod and Class are incompatible.
         """
         def err_message(message, *args):
-            location = '' if index is None else 'Error in element {0}: '.format(
-                index)
-            return ''.join(
-                (location,
-                 'PassMethod {0} is not compatible with '.format(pass_method),
-                 message.format(*args), '\n{0}'.format(kwargs)))
+            location = ': ' if index is None else ' {0}: '.format(index)
+            return ''.join(('Error in element', location,
+                            'PassMethod {0} '.format(pass_method),
+                             message.format(*args), '\n{0}'.format(kwargs)))
 
         pass_method = kwargs.get('PassMethod')
         if pass_method is not None:
             pass_to_class = PASS_MAPPING.get(pass_method)
             length = float(kwargs.get('Length', 0.0))
-            if (pass_method == 'IdentityPass') and (length != 0.0):
-                raise AttributeError(err_message("length {0}.", length))
+            extension = sysconfig.get_config_vars().get('EXT_SUFFIX', '.so')
+            file_path = os.path.realpath(os.path.join(__file__,
```

Comment:
This feels a bit brittle to me. Could you import the module instead, or is there a reason you don't want to do that?

Reply:
Whatever method we use we have to do all this messy filepath finding.
If we do a try except to open and read the file then we have to catch different errors, `IOError` in python 2 and `FileNotFoundError` in python 3. I just thought that was messier than `os.path.exists`.